src/vehicle.cpp

```diff
@@ -354,6 +354,7 @@ Vehicle::Vehicle(VehicleType type)
 	this->cargo_age_counter  = 1;
 	this->last_station_visited = INVALID_STATION;
 	this->last_loading_station = INVALID_STATION;
+	this->last_station_departure_date = INVALID_DATE;
```

Comment:
What value is most useful for NewGRF, when the vehicle is newly built and has not yet visited a station?
* 0 = infinitely in the past
* INVALID_DATe = infinitely in the future (used now)
* build-date (used by date_of_last_service)

Reply:
0 or INVALID_DATE are the most useful.  Does INVALID_DATE return an actual numeric value to newgrf?